cmd/controller-manager/app/options/options.go

```diff
@@ -35,6 +26,8 @@ type Options struct {
 	// to karmada-apiserver. In that case, be cautious when changing the constant, it must work with
 	// ClusterMonitorGracePeriod(--cluster-monitor-grace-period) in karmada-controller-manager.
 	ClusterStatusUpdateFrequency metav1.Duration
+	// LeaderElection determines whether or not to use leader election when starting the manager.
+	LeaderElection bool
```

Comment:
It would be incompatible change later on when we want to expose more configs for the leader election functionality since it's now just a bool

Reply:
yeah, which flags do you think we should expose according to `https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L156`

Reply:
Another option:
How about putting the `leader-elect` flag here?:
https://github.com/karmada-io/karmada/blob/c9f3fd8997e4b3c73f74bbefee630497feac039f/cmd/controller-manager/app/options/options.go#L27

Reply:
You may simply try with the following change:

```diff
--- a/cmd/controller-manager/app/options/options.go
+++ b/cmd/controller-manager/app/options/options.go
@@ -78,4 +51,6 @@ func (o *Options) AddFlags(flags *pflag.FlagSet) {
 		"The secure port on which to serve HTTPS.")
 	flags.DurationVar(&o.ClusterStatusUpdateFrequency.Duration, "cluster-status-update-frequency", 10*time.Second,
 		"Specifies how often karmada-controller-manager posts cluster status to karmada-apiserver.")
+	flags.BoolVar(&o.LeaderElection.LeaderElect, "leader-elect", true, "Start a leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability.")
+
 }
```

xref: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/

Reply:
ok